src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,161 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private final int defaultSize = 5;
+    private Object[] defaultNewCapacity;
+    private T[] elementArray;
+    private int size;
+
+    public ArrayList() {
+        elementArray = (T[]) new Object[defaultSize];
+    }
+
+    public ArrayList(int capacity) {
+        elementArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        elementArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(" Index "
+                    + index + " out of bounds for length " + size);
+        }
+        defaultNewCapacity = new Object[size + 1];
+        for (int i = 0; i < defaultNewCapacity.length; i++) {
+            if (i < index) {
+                defaultNewCapacity[i] = elementArray[i];
+            }
+            if (i == index) {
+                defaultNewCapacity[i] = value;
+            }
+            if (i > index) {
+                defaultNewCapacity[i] = elementArray[i - 1];
+            }
+        }
+        size++;
+        elementArray = (T[]) defaultNewCapacity;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() <= 0) {
+            throw new IndexOutOfBoundsException("list " + list.size()
+                    + " out of bounds for length " + size);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexSizeArray(index);
+        return elementArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexSizeArray(index);
+        elementArray[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexSizeArray(index);
+        defaultNewCapacity = new Object[size - 1];
+        T result = get(index);
+        for (int i = 0; i < defaultNewCapacity.length; i++) {
+            if (i < index) {
+                defaultNewCapacity[i] = elementArray[i];
+            }
+            if (result.equals(elementArray[i]) || i == index) {
+                defaultNewCapacity[i] = elementArray[i + 1];
+            }
+            if (i > index) {
+                defaultNewCapacity[i] = elementArray[i + 1];
+            }
+        }
+        size--;
+        elementArray = (T[]) defaultNewCapacity;
+        return result;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = findIndexElement(element);
+        if (index >= size || index < 0) {
+            throw new NoSuchElementException("Element " + element
+                    + " out of bounds for length " + size);
+        }
```

Comment:
use checkIndex method here

Reply:
i tried the checkIndex method throws an error it wants a NoSuchElementException